shared/src/ledger/native_vp/ibc/context.rs

```diff
@@ -237,6 +239,19 @@ where
         );
         self.write(&current_tx_key, record.serialize_to_vec())?;
         self.write(&head_tx_key, (current_tx_idx + 1).serialize_to_vec())?;
+        for description in shielded
```

Comment:
I see a bit of duplication in how we update the nullifier set in the transaction codes. But maybe getting rid of it is easier said than done and can be deferred to a later PR?

Reply:
I've tried to improve the thing with a shared function now